packages/x-data-grid-pro/src/hooks/features/rows/useGridRowAriaAttributes.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import {
+  GridTreeNode,
+  useGridSelector,
+  gridFilteredTopLevelRowCountSelector,
+  gridFilteredChildrenCountLookupSelector,
+  gridExpandedSortedRowTreeLevelPositionLookupSelector,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid';
+import { useGridRowAriaAttributes as useGridRowAriaAttributesCommunity } from '@mui/x-data-grid/internals';
+import { useGridPrivateApiContext } from '../../utils/useGridPrivateApiContext';
+import { useGridRootProps } from '../../utils/useGridRootProps';
+
+export const useGridRowAriaAttributes = (addTreeDataAttributes?: boolean) => {
+  const apiRef = useGridPrivateApiContext();
+  const props = useGridRootProps();
+  const getRowAriaAttributesCommunity = useGridRowAriaAttributesCommunity();
+
+  const filteredTopLevelRowCount = useGridSelector(apiRef, gridFilteredTopLevelRowCountSelector);
+  const filteredChildrenCountLookup = useGridSelector(
+    apiRef,
+    gridFilteredChildrenCountLookupSelector,
+  );
+  const sortedVisibleRowPositionsLookup = useGridSelector(
+    apiRef,
+    gridExpandedSortedRowTreeLevelPositionLookupSelector,
+  );
+
+  const addAttributes =
+    addTreeDataAttributes !== undefined ? addTreeDataAttributes : props.treeData === true;
+
+  return React.useCallback(
+    (rowNode: GridTreeNode, index: number) => {
+      const ariaAttributes = getRowAriaAttributesCommunity(rowNode, index);
+
+      if (!addAttributes || rowNode === null) {
+        return ariaAttributes;
+      }
+
+      ariaAttributes['aria-level'] = rowNode.depth + 1;
+
+      const filteredChildrenCount = filteredChildrenCountLookup[rowNode.id] ?? 0;
+      // aria-expanded should only be added to the rows that contain children
+      if (rowNode.type === 'group' && filteredChildrenCount > 0) {
+        ariaAttributes['aria-expanded'] = Boolean(rowNode.childrenExpanded);
+      }
+
+      // if the parent is null, set size and position cannot be determined
+      if (rowNode.parent !== null) {
+        ariaAttributes['aria-setsize'] =
```

Comment:
Should the pinned (aggregation) row be included in `aria-setsize` calculation?
In [this](https://deploy-preview-13623--material-ui-x.netlify.app/x/react-data-grid/#premium-plan) demo, the pinned row _does_ have an `aria-setsize` set but no `aria-posinset` set.

![image](https://github.com/user-attachments/assets/8aa6182d-cff3-4710-91e0-796ecf2df120)

Also the `aria-setsize` calculation does not include this pinned row.

I can see two options:
1. If the pinned row should not be part of the aria attributes calculation, pinned row should not have a `aria-setsize` attribute too.
2. Otherwise, the `aria-setsize` should be calculated including the pinned row and it should also have a `aria-posinset` (18 in this specific case instead of 17)

What do you think?

Reply:
I see that pinned and aggregation rows are placed outside of the `rowgroup` element.
In the docs it is stated
> aria-posinset property
Defines an element's number or position in the current set of listitems or treeitems.

Which means that the set related attributes should be removed (option 1)

Reply:
I will also add tests to cover this